### ISSUE TITLE
[MHV Portal / Accessibility] Explore accessible alternatives to print/download actions menu

### ISSUE THREAD

**Issue by sarahhortonadhoc** [department-of-veterans-affairs/va.gov-team#125496]
## Description

Explore approaches to providing a print/download feature, to address accessibility issues with the implementation currently in use on MHV. Approach the task with the secondary objectives of proposing a new actions menu experimental component to VADS, as well as establishing infrastructure and procedures to support a common set of components and patterns for MHV.

## Context

MHV provides a custom actions menu button for printing and downloading files. The component is not included in VADS and is implemented differently on different MHV products. In addition, some MHV products provide completely different controls and patterns for printing and downloading. Recent accessibility conformance reviews found several standards compliance defects. 

<img width="330" height="232" alt="Screenshot of expanded Print or download menu" src="https://github.com/user-attachments/assets/2cecba44-70cf-41d3-af01-2b17f6053880" /><br>
Expanded “Print or download” menu provided on [Medications list page](https://staging.va.gov/my-health/medications?page=1)

## User story

As a Veteran, I need interactive controls on the va.gov website to work with my interaction methods and strategies, so that I can access and operate the controls to complete my tasks.

As a Veteran, I need common features on the va.gov website to work in the same way, so that I can learn how to operate and apply what I know whenever I encounter the feature across the site.

### Patterns and examples
- [Menu Button Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/) from W3C ARIA Authoring Practices Guide (APG)
- [ARIA: menu role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/menu_role) from MDN
- [Menu buttons](https://carbondesignsystem.com/components/menu-buttons/usage/) from Carbon Design System
- TBC

### Accessibility-related tickets
- #97264 — from MHV Appointments
- #122375 — from MHV Medical Records and MHV Medications
- TBC

## Notes

- See a previous [Experimental Design [accordion menu print and download (list) button] #1877](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1877) ticket for details about a similar proposed component from Jun 2023

## Possible tasks

- [x] Create an interface inventory of print/download features across MHV and va.gov — see [inventory overview, below](https://github.com/department-of-veterans-affairs/va.gov-team/issues/125496#issuecomment-3549546792)
- [x] Compare a range of implementations, e.g., established design systems, comparable products and services — see [similar components, below](https://github.com/department-of-veterans-affairs/va.gov-team/issues/125496#issuecomment-3577742862)
- [x] Mock up possible design approaches — see [mockups, below](https://github.com/department-of-veterans-affairs/va.gov-team/issues/125496#issuecomment-3581682573)
- [ ] Create functional prototypes of candidate design approaches
- [ ] Assess the accessibility and usability of prototypes
- [ ] Choose an approach for implementation

## Acceptance criteria

- [ ] UX review by @fmccaf1 completed
- [ ] UX review by UX/design lead of tool teams completed
- [ ] Content review completed
- [ ] Accessibility review completed
- [ ] Feedback incorporated from stakeholders
- [ ] Follow-on implementation tickets created
- [ ] Design documentation completed (description, discovery learnings, design decisions)

**Comment by sarahhortonadhoc** [comment]
### Types of print features

#### Plain Button
<img width="94" height="49" alt="Screenshot of plain print button" src="https://github.com/user-attachments/assets/dab1d019-4c94-4a41-bbbb-0f1d6dce4f4b" />

#### Icon Button
<img width="244" height="53" alt="Screenshot of icon print button" src="https://github.com/user-attachments/assets/a677b431-d0b2-423c-8ab2-0dfc373f0297" />

#### Action Menu Button (expanded)
<img width="330" height="232" alt="Screenshot of expanded print or download action menu button" src="https://github.com/user-attachments/assets/075f64d4-1808-4369-8025-a357a25b9371" />

### Anatomy of print features

|Element           |Item |Label/Value                                     |Source|
| :--- | :--- | :--- | :--- |
|Plain Button      |Label|Print                                           |VADS  |
|Icon Button       |Icon |Printer                                         |Custom|
|                  |Label|Print                                           |      |
|Action Menu Button|Label|Print or download                               |Custom|
|                  |Icon |Expand                                          |      |
|                  |Icon |Collapse                                        |      |
|                  |Label|Print this [list][page]                         |      |
|                  |Label|Print PDF of this [list][page]                  |      |
|                  |Label|Download a text file (.txt) of this [list][page]|      |

### Use of print features on MHV products

| Product      | Page                 | Feature        | Component           |
| :--- | :--- | :--- | :--- |
| Appointments | List                 | Print          | Icon Button         |
| Appointments | Detail               | Print          | Plain Button        |
| Messages     | List                 | None           | None                |
| Messages     | Detail               | Print          | Icon Button (Group) |
| Medications  | List                 | Print/download | Action Menu Button  |
| Medications  | Detail               | Print/download | Action Menu Button  |
| Records      | List                 | None           | None                |
|              | List (Vaccines only) | Print/download | Action Menu Button  |
| Records      | Detail               | Print/download | Action Menu Button  |

**Comment by sarahhortonadhoc** [comment]
Moved to blocked following the [Dec 22 presentation at MHV Patient Portal Design CoP Sync](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/digital-health-modernization/mhv-to-va.gov/accessibility/README.md#presentations) — waiting for further discussion and guidance (see [Slack thread](https://dsva.slack.com/archives/C04DRS3L9NV/p1766503429025179)).

**Comment by sarahhortonadhoc** [comment]
Follow-up on next steps to move from blocked (see [#mhv-accessibility Slack thread](https://dsva.slack.com/archives/C05QG2H510T/p1767031211159279).
